code/backend/batch/utilities/orchestrator/prompt_flow.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import List
+import urllib.request
+import json
+import tempfile
+
+from .orchestrator_base import OrchestratorBase
+from ..common.answer import Answer
+from ..helpers.llm_helper import LLMHelper
+
+logger = logging.getLogger(__name__)
+
+
+class PromptFlowOrchestrator(OrchestratorBase):
+    def __init__(self) -> None:
+        super().__init__()
+        self.llm_helper = LLMHelper()
+
+        # Get the ML client and endpoint name
+        self.ml_client = self.llm_helper.get_ml_client()
+        self.enpoint_name = self.llm_helper.get_endpoint_name()
+        self.deployment_name = self.llm_helper.get_deployment_name()
+
+    async def orchestrate(
```

Comment:
Could we maybe break this method into several smaller ones? It should make it easier to see the different steps involved, but also would allow us to remove some of the inline comments.

Reply:
I popped out the data into file transformation to a separate method but thought the rest made sense to stay in orchestrate especially that it follows the same orchestrate format as lang chain, semantic kernel and open ai functions orchestrators. Happy to discuss and make further changes if you think this is still not very readable.

Reply:
broke out the transform_data_into_file method. rest left as is as this is the same structure followed by lang chain, semantic kernel and open ai functions orchestrators.